src/main.rs

```diff
@@ -808,6 +896,16 @@ fn main() {
         return;
     }
 
+    if let Err(e) = expand_fdtable().map_err(Error::ExpandFdTable) {
```

Comment:
Should we skip this if we know we won't be restoring?  e.g. if a VM is started immediately via command line flags.

Reply:
We do consume a lot of file descriptors during normal boot (through cloning eventfd file descriptors) so it seems useful beyond the restore case.

Reply:
The fix should improve the startup time across all scenarios, not just the restore code path. Once you have more than one thread and you use more than 64 file handles in your process you'll encounter this issue.

Reply:
Makes sense.